ext/reflection/php_reflection.c

```diff
@@ -2809,19 +2809,25 @@ ZEND_METHOD(reflection_parameter, getDefaultValueConstantName)
 
 	param = _reflection_param_get_default_param(INTERNAL_FUNCTION_PARAM_PASSTHRU);
 	if (!param) {
-		return;
+		RETURN_THROWS();
+	}
+
+	zval default_value_zval;
+	if (_reflection_param_get_default_arg_info(param->arg_info, &default_value_zval) == FAILURE) {
+		RETURN_THROWS();
 	}
 
-	precv = _reflection_param_get_default_precv(INTERNAL_FUNCTION_PARAM_PASSTHRU, param);
-	if (precv && Z_TYPE_P(RT_CONSTANT(precv, precv->op2)) == IS_CONSTANT_AST) {
-		zend_ast *ast = Z_ASTVAL_P(RT_CONSTANT(precv, precv->op2));
+	if (Z_TYPE(default_value_zval) == IS_CONSTANT_AST) {
+		zend_ast *ast = Z_ASTVAL(default_value_zval);
 
 		if (ast->kind == ZEND_AST_CONSTANT) {
-			RETURN_STR_COPY(zend_ast_get_constant_name(ast));
+			RETVAL_STR_COPY(zend_ast_get_constant_name(ast));
 		} else if (ast->kind == ZEND_AST_CONSTANT_CLASS) {
-			RETURN_STRINGL("__CLASS__", sizeof("__CLASS__")-1);
+			RETVAL_STRINGL("__CLASS__", sizeof("__CLASS__")-1);
 		}
 	}
+
+	zval_dtor(&default_value_zval);
```

Comment:
We should add a new API like `getDefaultValueString()` or `getDefaultValueExpression()` and deprecate `isDefaultValueConstant()` and `getDefaultValueConstantName()` in favor of that.

Reply:
I see the point, and it makes sense (although I am a little bit unhappy that I have to add back the `...NO_CONSTANT_SUBSTITUTION` flags 😕 😄)

Reply:
Why do you need to add back the flags?

Reply:
First I thought that deprecating means "leaving alone": so that I should revert the changes made to `isDefaultValueConstant()` and `getDefaultValueConstantName()`, and only add the new behaviour to the new methods. In the meanwhile I noticed that it's not the case.